packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -150,21 +267,40 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
     this.match = options.match;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
-    }
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    const pathMatchConfig = (this.match?.path ?? HttpGatewayRoutePathMatch.startsWith('/')).bind(scope);
+    const defaultTargetHostname = this.match?.rewriteRequestHostname;
+
+    // Set prefix path match to '/' if none of path matches are defined.
+    const prefixPathRewrite = pathMatchConfig?.prefixPathRewrite;
+    const wholePathRewrite = pathMatchConfig?.wholePathRewrite;
```

Comment:
```suggestion
    const prefixPathRewrite = pathMatchConfig.prefixPathRewrite;
    const wholePathRewrite = pathMatchConfig.wholePathRewrite;
```

(`pathMatchConfig` is never `undefined`)